src/base/http/requestparser.cpp

```diff
@@ -310,26 +337,28 @@ bool RequestParser::parseFormData(const QByteArrayView data)
         return false;
     }
 
-    const QString headers = QString::fromLatin1(data.first(eohPos));
+    QByteArrayView headers = data.first(eohPos);
     const QByteArrayView payload = viewWithoutEndingWith(data.sliced((eohPos + EOH.size())), CRLF);
 
     HeaderMap headersMap;
-    const QList<QStringView> headerLines = QStringView(headers).split(QString::fromLatin1(CRLF), Qt::SkipEmptyParts);
-    for (const auto &line : headerLines)
+    while (!headers.isEmpty())
     {
-        if (line.trimmed().startsWith(HEADER_CONTENT_DISPOSITION, Qt::CaseInsensitive))
+        QByteArrayView line = splitView(headers, CRLF);
+        if (line.isEmpty()) continue;
+
+        if (QString::fromLatin1(line.trimmed()).startsWith(HEADER_CONTENT_DISPOSITION, Qt::CaseInsensitive))
         {
             // extract out filename & name
-            const QList<QStringView> directives = line.split(u';', Qt::SkipEmptyParts);
-
-            for (const auto &directive : directives)
+            while (!line.isEmpty())
             {
-                const int idx = directive.indexOf(u'=');
-                if (idx < 0)
-                    continue;
+                QByteArrayView directive = splitView(line, QByteArrayLiteral(";"));
+                if (directive.isEmpty()) continue;
+
+                const QByteArrayView part = splitView(directive, QByteArrayLiteral("="));
+                if (directive.isEmpty()) continue;
 
-                const QString name = directive.first(idx).trimmed().toString().toLower();
-                const QString value = Utils::String::unquote(directive.sliced(idx + 1).trimmed()).toString();
+                const QString name = QString::fromLatin1(part.trimmed()).toLower();
+                const QString value = QString::fromUtf8(unquotedView(directive.trimmed()));
```

Comment:
I believe we can add this comment:
```suggestion
                // [rfc7578] 4.2. Content-Disposition Header Field for Each Part
                const QString value = QString::fromUtf8(unquotedView(directive.trimmed()));
```
Quoting the relevant part:
>Some commonly deployed systems use multipart/form-data with file
   names directly encoded including octets outside the US-ASCII range.
   The encoding used for the file names is typically UTF-8, although
   HTML forms will use the charset associated with the form.